include/z64collision_check.h

```diff
@@ -9,245 +9,345 @@
 } ColliderJntSphInit; // size = 0x10
 
 typedef struct {
     /* 0x00 */ ColliderInit_Set3 base;
     /* 0x08 */ s32 count;
-    /* 0x0C */ ColliderJntSphItemInit* list;
+    /* 0x0C */ ColliderJntSphElementInit* elements;
 } ColliderJntSphInit_Set3; // size = 0x10
 
 typedef struct {
-    /* 0x00 */ ColliderInit_Actor base;
+    /* 0x00 */ ColliderInitToActor base;
     /* 0x08 */ s32 count;
-    /* 0x0C */ ColliderJntSphItemInit* list;
-} ColliderJntSphInit_Actor; // size = 0x10
+    /* 0x0C */ ColliderJntSphElementInit* elements;
+} ColliderJntSphInitToActor; // size = 0x10
 
 typedef struct {
     /* 0x00 */ Collider base;
-    /* 0x18 */ ColliderBody body;
+    /* 0x18 */ ColliderInfo info;
     /* 0x40 */ Cylinder16 dim;
 } ColliderCylinder; // size = 0x4C
 
 typedef struct {
     /* 0x00 */ ColliderInit base;
-    /* 0x08 */ ColliderBodyInit body;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ Cylinder16 dim;
 } ColliderCylinderInit; // size = 0x2C
 
 typedef struct {
     /* 0x00 */ ColliderInit_Set3 base;
-    /* 0x08 */ ColliderBodyInit body;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ Cylinder16 dim;
 } ColliderCylinderInit_Set3; // size = 0x2C
 
 typedef struct {
-    /* 0x00 */ ColliderInit_Actor base;
-    /* 0x08 */ ColliderBodyInit body;
+    /* 0x00 */ ColliderInitToActor base;
+    /* 0x08 */ ColliderInfoInit info;
     /* 0x20 */ Cylinder16 dim;
-} ColliderCylinderInit_Actor; // size = 0x2C
+} ColliderCylinderInitToActor; // size = 0x2C
 
 typedef struct {
     /* 0x00 */ Vec3f vtx[3];
-} ColliderTrisItemDimInit; // size = 0x24
+} ColliderTrisElementDimInit; // size = 0x24
 
 typedef struct {
-    /* 0x00 */ ColliderBody body;
+    /* 0x00 */ ColliderInfo info;
     /* 0x28 */ TriNorm dim;
-} ColliderTrisItem; // size = 0x5C
+} ColliderTrisElement; // size = 0x5C
 
 typedef struct {
-    /* 0x00 */ ColliderBodyInit body;
-    /* 0x18 */ ColliderTrisItemDimInit dim;
-} ColliderTrisItemInit; // size 0x3C
+    /* 0x00 */ ColliderInfoInit info;
+    /* 0x18 */ ColliderTrisElementDimInit dim;
+} ColliderTrisElementInit; // size 0x3C
 
 typedef struct {
     /* 0x00 */ Collider base;
     /* 0x18 */ s32 count;
-    /* 0x1C */ ColliderTrisItem* list;
+    /* 0x1C */ ColliderTrisElement* elements;
 } ColliderTris; // size = 0x20
 
 typedef struct {
     /* 0x00 */ ColliderInit base;
     /* 0x08 */ s32 count;
-    /* 0x0C */ ColliderTrisItemInit* list;
+    /* 0x0C */ ColliderTrisElementInit* elements;
 } ColliderTrisInit; // size = 0x10
 
 typedef struct {
     /* 0x00 */ ColliderInit_Set3 base;
     /* 0x08 */ s32 count;
-    /* 0x0C */ ColliderTrisItemInit* list;
+    /* 0x0C */ ColliderTrisElementInit* elements;
 } ColliderTrisInit_Set3; // size = 0x10
 
 typedef struct {
     /* 0x00 */ Vec3f quad[4];
     /* 0x30 */ Vec3s dcMid; // midpoint of vectors d, c
     /* 0x36 */ Vec3s baMid; // midpoint of vectors b, a
-    /* 0x3C */ f32 unk_3C;
+    /* 0x3C */ f32 acDist;
```

Comment:
could you add a comment on what this is?

Reply:
Player quads use it to keep track of which collider element they hit was closest to them and only apply hit effects to that one. If a player quad has an AC collision with an element closer than acDist, it stores the new dist to acDist and resets the previous collider.